### PR TITLE
DCOS-13422: Fix TaskTable's status sorting method

### PR DESCRIPTION
**Pull request by jfurrow**
Before:

After: 

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

**Comment by jfurrow** [comment]
@leemunroe Can you please verify that the ordering of these states is correct? I chose them based on what I think is the most important... Here are the values (lower values mean they will be higher in the list when in ascending order): https://github.com/dcos/dcos-ui/blob/ae468987c3fbaeef9c259e94abadeaeb739870d9/plugins/services/src/js/constants/TaskStatusSorting.js

File: foundation-ui/utils/ResourceTableUtil.js [hunk 1]
```diff
@@ -45,6 +46,8 @@ var ResourceTableUtil = {
 
   getSortFunction(tieBreakerProp) {
```
Comment (nLight) [inline foundation-ui/utils/ResourceTableUtil.js]: How about a small refactoring? The name of the file is `ResourceTableUtil` but `ResourceTable` does't exist. That's ok, the `getSortFunction` is being used only by the `TaskTable` if I'm not mistaken. Let's extract this function into `TaskTableUtil`. Then I won't be doing a hardcore soul searching justifying presence of a specific constant `TaskStatusSorting` inside a generic function `getSortFunction`. Does it make sense? 

Happy to discuss this today.

Comment (jfurrow) [inline foundation-ui/utils/ResourceTableUtil.js]: Good idea! Will work on this now :)

File: plugins/services/src/js/constants/TaskStatusSorting.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const TaskStatusSorting = {
```
Comment (jfurrow) [inline plugins/services/src/js/constants/TaskStatusSorting.js]: Yes, that's a much better name. Done!

File: foundation-ui/utils/ResourceTableUtil.js (hunk 1 above)
Comment (jfurrow) [inline foundation-ui/utils/ResourceTableUtil.js]: Done!

**Comment by nLight** [comment]
run unit tests

**Comment by nLight** [comment]
John you are awesome!

giphy-2

**Comment by jfurrow** [comment]
No *you're* awesome!

**Comment by rcorral** [comment]
Works great, thanks @jfurrow!!

**Pull request merged**